pkg/api/mlflow/config/config.go

```diff
@@ -0,0 +1,35 @@
+package config
+
+import (
+	"time"
+
+	"github.com/spf13/viper"
+)
+
+// ServiceConfig represents main service configuration.
+type ServiceConfig struct {
+	AuthUsername          string
+	AuthPassword          string
+	ArtifactRoot          string
+	S3EndpointURL         string
+	DatabaseURI           string
+	DatabaseReset         bool
+	DatabasePoolMax       int
+	DatabaseMigrate       bool
+	DatabaseSlowThreshold time.Duration
+}
+
+// NewServiceConfig creates new instance of ServiceConfig.
+func NewServiceConfig() *ServiceConfig {
+	return &ServiceConfig{
+		AuthUsername:          viper.GetString("auth-username"),
```

Comment:
can't viper automatically populate a config struct that has annotations?

Reply:
this is a good question. I have to check.

Reply:
ok, I did some sort of investigation. Viper supports reading configuration to `struct` -> https://github.com/spf13/viper#unmarshaling but a bit strange. I can't see that for `default` values for example. So I think I need more time to play with that so Im going to create a task and put it into `backlog` to investigate deeper a bit later.

Reply:
https://github.com/G-Research/fasttrackml/issues/154